src/bitbucket.js

```diff
@@ -0,0 +1,88 @@
+const fetch = require('node-fetch');
+const FormData = require('form-data');
+const { fetch_upload_data } = require('./utils');
+const {Base64} = require('js-base64');
+
+const {
+  BITBUCKET_REPO_FULL_NAME = '', //namespace with project name, i.e. elle/cml
+  BITBUCKET_REPO_UUID, // UUID of project- string in {}
+  BITBUCKET_BRANCH, // branch, cannot do build against tag
+  BITBUCKET_TOKEN, // token, cannot do build against branch
+  BITBUCKET_COMMIT, // SHA of commit
+  BITBUCKET_PR_ID, // ID of pull request- only available if build is triggered on PR
+  // GITLAB_USER_EMAIL, // doesn't look like email is part of BB env vars
+  BITBUCKET_WORKSPACE, //username
+  BITBUCKET_GIT_HTTP_ORIGIN, // url to project
+  BITBUCKET_REPO_SLUG, // repo slug
+  repo_token
+} = process.env;
+
+
+const USERNAME = BITBUCKET_WORKSPACE;
+const IS_PR = BITBUCKET_PR_ID;
+const REF = BITBUCKET_BRANCH || BITBUCKET_TOKEN;
+const PASSWORD = repo_token;
+const API_URL = "https://api.bitbucket.org/2.0";
+
+const comment = async (opts) => {
+  const { commit_sha, report } = opts;
+
+  const endpoint = `/repositories/${USERNAME}/${BITBUCKET_REPO_SLUG}/commit/${BITBUCKET_COMMIT}/comments/`;
```

Comment:
should be using commit_sha, the cml command will set it with BITBUCKET_COMMIT by default since you are exposing BITBUCKET_COMMIT later on as head_sha ;)